### PR TITLE
Lis histogram element

### PR DESCRIPTION
**Pull request by ctcncgr**
Adds a component to use for histograms.

See:

http://dev.lis.ncgr.org:50050/dev/core/lis-histogram-element.html

Docs:

http://dev.lis.ncgr.org:50050/docs/classes/core_components.LisHistogramElement.html

### CODE REVIEW

**Comment by ctcncgr** [comment]
No rush @adf-ncgr and @alancleary as you guys have time to review this. I'm going to move on to other views in the meantime. DSCensor replace inc...

File: src/core/lis-histogram-element.ts [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+      .attr(
+        'transform',
+        'translate(' + width / 2 + ' ,' + (height + padding - 5) + ')',
+      )
+      .style('text-anchor', 'middle')
+      .style('fill', 'black')
+      .text(this.orientation === 'vertical' ? this._ylabel : this._xlabel);
+
+    // Add the y-axis label
+    svgContainer
```
Comment (alancleary) [inline src/core/lis-histogram-element.ts:240]: The edge of the yAxis label is being trimmed in the example. Can this be adjusted or the space allocated dynamically?

**Review by alancleary — CHANGES_REQUESTED** [review]
Overall I think this is a solid start. There's a handful of minor changes that need to be made; see my code comments for details.

Two things I want to discuss here are:

1. The size of the component is completely independent of the component's container so I don't see the need for the ResizeObserver. I think it would be cool if the width/height of the histogram were drawn to be the width/height of the container if a width/weight isn't explicitly specified. For example, you would give an explicit width but leave the height empty so the height would be the size of the container. Alternatively, it might make sense to let the "name" axis to be as large as it needs to be to fit the given data.
2. I think having `xlabel` and `ylabel` properties is confusing because the orientation switches what axis the labels are in. Would it make sense instead to name the label by the data on their axis? For example, `nameLabel` and `countLabel`.

File: src/core/lis-histogram-element.ts [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+ * </lis-histogram-element>
+ * ```
+ */
+@customElement('lis-histogram-element')
+export class LisHistogramElement extends LitElement {
+  // bind to the histogram container div element in the template
+  private _histogramContainerRef: Ref<HTMLDivElement> = createRef();
+
+  // a controller that allows element resize events to be observed
+  protected resizeObserverController = new LisResizeObserverController(
```
[the remark this answers is not in the document]
Comment (ctcncgr) [reply src/core/lis-histogram-element.ts:72]: I had originally tested out putting the svg in a sizable div, but it ended up being buggy. I forgot to remove this. When I add it back I'll put this back in.

File: dev/core/lis-histogram-element.html [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  <script src="../../node_modules/uikit/dist/js/uikit.min.js"></script>
+  <!-- web components polyfills -->
+  <script src="../../node_modules/@webcomponents/webcomponentsjs/webcomponents-loader.js"></script>
+  <script src="../../node_modules/lit/polyfill-support.js"></script>
+  <!-- web components -->
+  <script type="module" src="../../lib/index.js"></script>
+</head>
+
+<body>
+  <div class="uk-container" style="overflow: visible;">
```
[the remark this answers is not in the document]
Comment (ctcncgr) [reply dev/core/lis-histogram-element.html:19]: This is a mistake that was leftover. It's not actually in the example code anymore on the example site. Will Fix.

**Comment by ctcncgr** [comment]
Closing this. Will resubmit with changes so its easier to track.

**Pull request closed without merging**